core/src/SYCL/Kokkos_SYCL_GraphNodeKernel.hpp

```diff
@@ -131,11 +131,11 @@ class GraphNodeKernelImpl<Kokkos::SYCL, PolicyType, Functor, PatternTag,
   }
 
  private:
-  Kokkos::ObservingRawPtr<sycl::ext::oneapi::experimental::command_graph<
-      sycl::ext::oneapi::experimental::graph_state::modifiable>>
-      m_graph_ptr = nullptr;
-  Kokkos::ObservingRawPtr<std::optional<sycl::ext::oneapi::experimental::node>>
-      m_graph_node_ptr = nullptr;
+  sycl::ext::oneapi::experimental::command_graph<
+      sycl::ext::oneapi::experimental::graph_state::modifiable>* m_graph_ptr =
+      nullptr;
+  std::optional<sycl::ext::oneapi::experimental::node>* m_graph_node_ptr =
```

Comment:
Let me check why this is a pointer to an optional. I don't mean to request a change in this PR but it might deserve a followup.

Reply:
OK, so for the `GraphNodeThenHostImpl` a bit up in the file, I've just used
```suggestion
  std::optional<sycl::ext::oneapi::experimental::node> m_node =std::nullopt;
```

Reply:
We already have a pointer to an optional.  We are not changing this in this PR.

Reply:
I don't love it either but I remember Daniel wanted this for some reason.

Reply:
https://github.com/kokkos/kokkos/pull/6912#discussion_r1617968503

Reply:
The [comment says](https://github.com/kokkos/kokkos/pull/6912#discussion_r1617968503):
> So you would prefer pointer to optional to resemble the CUDA/HIP code?
I suppose that's fine.

But I am not convinced by it. If I were to make a followup for the `hipGraphNode_t` (and `Cuda`), is there any strong reason for keeping the pointer to optional here, except "consistency with other specializations" ?

Reply:
~~I guess the reason for keeping a pointer the the optional graph is to shrink the size of the type.~~
It needs a reference to the graph, not the graph itself.